### PR TITLE
Survey validation enhancements

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: app/org/sagebionetworks/bridge/validators/SurveyValidator.java [hunk 1]
```diff
@@ -161,37 +169,91 @@ private void doValidateConstraints(SurveyQuestion question, Constraints con, Err
...
+    
+    private void doValidateConstraintsType(Errors errors, UIHint hint, DateConstraints con) {
+        LocalDate earliestDate = con.getEarliestValue();
+        LocalDate latestDate = con.getLatestValue();
+        if (earliestDate != null && latestDate != null) {
+            if (latestDate.isBefore(earliestDate)) {
+                rejectField(errors, "earliestValue", "is after the latest value");
+            }
+        }
+    }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/validators/SurveyValidator.java:223]: Unit test for validating DateConstraints?

File: app/org/sagebionetworks/bridge/validators/SurveyValidator.java [hunk 2]
```diff
@@ -161,37 +169,91 @@ private void doValidateConstraints(SurveyQuestion question, Constraints con, Err
...
+    
+    private void doValidateConstraintsType(Errors errors, UIHint hint, DateTimeConstraints con) {
+        DateTime earliestDate = con.getEarliestValue();
+        DateTime latestDate = con.getLatestValue();
+        if (earliestDate != null && latestDate != null) {
+            if (latestDate.isBefore(earliestDate)) {
+                rejectField(errors, "earliestValue", "is after the latest value");
+            }
+        }
+    }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/validators/SurveyValidator.java:233]: Unit test for validating DateTimeConstraints?

File: app/org/sagebionetworks/bridge/validators/SurveyValidator.java [hunk 3]
```diff
@@ -161,37 +169,91 @@ private void doValidateConstraints(SurveyQuestion question, Constraints con, Err
...
+            doValidateConstraintsType(errors, hint, (StringConstraints)con);
+        } else if (con instanceof DateConstraints) {
+            doValidateConstraintsType(errors, hint, (DateConstraints)con);
+        } else if (con instanceof DateTimeConstraints) {
+            doValidateConstraintsType(errors, hint, (DateTimeConstraints)con);
+        } else if (con instanceof NumericalConstraints) {
+            doValidateConstraintsType(errors, hint, (NumericalConstraints)con);
+        } else if (con instanceof StringConstraints) {
+            doValidateConstraintsType(errors, hint, (StringConstraints)con);
+        }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/validators/SurveyValidator.java:181]: Duplicate else-if for (con instanceof StringConstraints)

**Comment by DwayneJengSage** [comment]
A few minor issues, otherwise looks good.

**Comment by alxdarksage** [comment]
Added tests, removed redundant else/if clause.

**Pull request merged**